spec/factories/dialog.rb

```diff
@@ -1,6 +1,15 @@
 FactoryGirl.define do
   factory :dialog do
+    # HACK: This is required because we were previously depending on rspec-mocks'
+    # .stub monkeypatch here; the monkeypatch has since been removed and rspec-mocks
+    # should NOT be used within factories anyway.
+    # TODO: Rewrite dialog specs to properly mock this within the spec
     # skip validate_children callback for general dialog testing
-    before(:create) { |d| d.stub(:validate_children) }
+    to_create do |instance|
+      class << instance
+        def validate_children; true; end
+      end
+      instance.save!
+    end
```

Comment:
Note these changes (same thing in `dialog_group` and `dialog_tab`) in the sea of vanilla conversions of this large diff.